plugins/modules/lan_automation_workflow_manager.py

```diff
@@ -1316,20 +1380,39 @@ def validate_discovery_devices(self, discovery_devices, missing_params):
             - It validates the 'device_site_name_hierarchy' if provided, ensuring the site exists. If the site is
               invalid, it raises an error and halts execution.
         """
+        if discovery_devices is None:
+            self.log("No discovery devices found. Skipping validation for 'discovery_devices'.", "INFO")
+            return
 
         if discovery_devices:
```

Comment:
As we already checked   in line number 1383: 
`if discovery_devices is None:
`
we don't need this check in 1387: 
`if discovery_devices:`